docs/building-rpm-ostree-and-autosd.md

```diff
@@ -0,0 +1,171 @@
+<!-- markdownlint-disable-file MD013 MD033 -->
+# Building rpm-ostree in autosd image from scratch
+
+The AutoSD is an upstream repository for Red Hat In-Vehicle OS, much like CentOS Stream is to RHEL. AutoSD is based on CentOS Stream with a few divergences. If your base system is not a CentOS, you can use a container to generate a compatible CentOS Stream 9 RPM as showed below, otherwise just skip the step and use your raw system.
+
+## Building rpm-ostree rpm on CentOS Stream9
+
+Create a **ContainerFile** with the content below.
+
+```bash
+FROM centos:stream9
+
+RUN dnf update -y
+RUN dnf install -y 'dnf-command(config-manager)' epel-release
+RUN dnf config-manager --set-enabled crb
+RUN dnf config-manager --add-repo https://buildlogs.centos.org/9-stream/automotive/aarch64/packages-main/
+RUN dnf config-manager --add-repo https://buildlogs.centos.org/9-stream/autosd/aarch64/packages-main/
+RUN if [ $(uname -m) == "aarch64" ]; then dnf install -y --nogpgcheck autosig-u-boot; fi
+RUN dnf install -y --allowerasing --nogpgcheck mock cpio qemu-kvm rpm-build autoconf \
+  automake bison dracut git gpgme-devel gtk-doc libattr-devel libcap-devel libtool \
+  make openssl-devel fuse libarchive libselinux libsoup zlib \
+  gobject-introspection-devel e2fsprogs-devel fuse-devel libarchive-devel \
+  libcurl-devel libsoup-devel systemd-devel xz-devel flex python3-devel swig bc \
+  centos-release-automotive audit-libs-devel bzip2-devel cryptsetup-devel dbus-devel \
+  elfutils-devel firewalld-filesystem gcc-c++ gettext gnu-efi gnu-efi-devel \
+  gnutls-devel gperf hostname kmod-devel libacl-devel libfdisk-devel libgcrypt-devel \
+  libidn2-devel libmicrohttpd-devel libseccomp-devel libxkbcommon-devel \
+  libzstd-devel lz4 lz4-devel pam-devel perl tree valgrind-devel meson \
+  bash-completion tpm2-tss-devel attr dtc
+
+# Install deps required for recent rpm-ostree
+RUN cd rpmbuild/SPECS/ && curl -OL https://gitlab.com/redhat/centos-stream/rpms/rpm-ostree/-/raw/c9s/rpm-ostree.spec
+RUN cd /tmp && dnf builddep -y rpm-ostree.spec
+
+# Lets use the latest from community (see next curl rustup step).
+# Also avoids "cargo metadata: error: failed to run rustc to learn about target-specific information"
```

Comment:
Really?  I haven't hit that...seems strange

Reply:
yes, unfortunately, got it in several times in CentOS 9. :(

Reply:
Ahhh.  OK, I finally tried this (would have been helpful to see the full error)

https://github.com/coreos/cargo-vendor-filterer/issues/87